src/GovUk.Education.ExploreEducationStatistics.Data.Services/MetaGuidanceSubjectService.cs

```diff
@@ -95,6 +95,7 @@ public async Task<TimePeriodLabels> GetTimePeriods(Guid subjectId)
                 .Observation
                 .Where(observation => observation.SubjectId == subjectId)
                 .Select(observation => new {observation.Year, observation.TimeIdentifier})
+                .ToList()
```

Comment:
Seems fine but maybe `MetaGuidanceSubjectService` shouldn't get too hung up about this and just defer to `TimePeriodService` which already has a better version of this query in it, doing a distinct before returning from the database so that all the Year/TimeIdentifier combinations aren't pulled into memory before the ordering.

This list will be as long as the number of observations in the Subject when it doesn't need to be to get the first and last.

So you could consider moving this to `TimePeriodService` as something like `GetTimePeriodLabels` which would be similar to `GetTimePeriodRange` and reuse the private method `GetDistinctObservationTimePeriods` which doesn't have this bug.

After this we will only have this logic in one place.

It seems like that suffered from the same fate until EES-2429